test/cli/fixtures/hard-error-in-test.js

```diff
@@ -0,0 +1,5 @@
+QUnit.test( "contains a hard error", assert => {
+	assert.async(); // and ensure it can recover
```

Comment:
I didn't quite get this line. I assume it was needed to cover a certain branch, but it's not clear to me what it covers from an end-user perspective. Are hard errors from would-be async tests special? If so, perhaps make that part of the file/test name.